ktor-server/ktor-server-core/src/io/ktor/features/ContentNegotiation.kt

```diff
@@ -53,11 +44,22 @@ class ContentNegotiation(val converters: List<ContentConverterRegistration>) {
             pipeline.sendPipeline.intercept(ApplicationSendPipeline.Render) {
                 if (subject is OutgoingContent) return@intercept
 
-                val suitableConverters = acceptedTypes().mapNotNull { (contentType, _) ->
+                val suitableConverters = call.request.acceptItems().mapNotNull { (contentType, _) ->
                     feature.converters.firstOrNull { it.contentType.match(contentType) }
                 }
 
-                val converted = suitableConverters.mapNotNull { it.converter.convertForSend(this, subject) }.firstOrNull()
+                val charset = call.request.acceptCharsetItems().mapNotNull { (charset, _) ->
+                    when {
+                        charset == "*" -> Charsets.UTF_8
+                        Charset.isSupported(charset) -> Charset.forName(charset)
```

Comment:
too expensive, need to find first supported, don't need to map every provided charset, consider use `firstOrNull { }` and then convert single value